R/plotLoadings.R

```diff
@@ -0,0 +1,374 @@
+#'   (default: \code{n = 10})
+#'   
+#' @param ncomponents A numeric specifying the number of components.
+#'   (default: \code{ncomponents = 5})
+#' 
+#' @param tree_name a single \code{character} value specifying a rowTree/colTree from
+#'   \code{object}. (By default: \code{tree_name = "phylo"})
+#'     
+#' 
+#' @details
+#' It is impossible to plot tree if only the matrix is given. Number of features must be reduced
+#' before calling function or it will not be understandable. 
+#' 
+#' @return 
+#' A \code{ggplot2} object. A circular plot annotated with TreeSummarizedExperiment object.
+#'
+#' @name plotLoadings
+#' @export
+#' 
+#' @author
+#' Ely Seraidarian
+#' Contact: \url{microbiome.github.io}
+#' @examples
+#' 
+#' 
+#' # Plotting feature loadings with tree
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- transformAssay(tse, method = "clr", pseudocount = 1)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", update.tree = TRUE)
+#' tse <- runPCA(tse, ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse, dimred= "PCA", layout = "tree")
+#' 
+#' # Plotting without tree as a heatmap
+#' loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+#' plotLoadings(loadings_matrix, layout = "heatmap")
+#' 
+#' # Plotting without tree as a barplot
+#' plotLoadings(loadings_matrix, layout = "barplot")
+#' 
+#' # Plotting without tree as a screeplot
+#' plotLoadings(loadings_matrix, layout = "screeplot")
+#' 
+#' # Plotting more features
+#' plotLoadings(loadings_matrix, layout = "heatmap", n = 12)
+#'
+#' # Plotting with less components
+#' tse <- runPCA(tse, ncomponents = 4, assay.type = "clr")
+#' loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+#' plotLoadings(loadings_matrix, layout = "heatmap", ncomponents = 4 )
+#' 
+#' # Plotting if loadings matrix name has been changed
+#' tse <- runPCA(tse, name = "myPCAmatrix", ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse, dimred= "myPCAmatrix", layout = "heatmap")
+NULL
+
+#' @rdname plotLoadings
+setGeneric("plotLoadings", signature = c("x"),
+    function(x, ...) 
+        standardGeneric("plotLoadings"))
+
+
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "TreeSummarizedExperiment"),
+    function(x,
+            dimred = "PCA",
+            layout = "heatmap",
+            n = 10,
+            ncomponents = 5,
+            tree_name = "phylo",
+            ...) {
+        
+            
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                        dimred = dimred,
+                        layout = layout,
+                        n = n,
+                        ncomponents = ncomponents,
+                        tree_name = tree_name,
+                        ...)  
+
+        loadings_matrix <- attr(reducedDim(x, dimred), "rotation")
```

Comment:
This works now with PCA. How about with RDA (mia::getRDA)

Reply:
This does not work for now, but @thpralas will create a wrapper for every reducedDim methods and store in feature_loadings instead of rotation so it will work for every method. I tried and currently RDA loadings matrix is not stored in rotation slot so it will not work here

Reply:
Wrappers might not be optimal solutions since it requires more maintenance and it might be that users still use scater::runPCA instead of our wrapper. It might be then quite confusing for user why the function does not work even though user used the "correct" function.

Instead, you could create an internal function

```
loading_names <- c("rotation", "loadings", "some_other_name")
attr_names <- names(attributes(reduced_dim))
attr_name <- attr_names[ attr_names %in% loading_names ]
if( length(attr_name) != 1 )){
    stop("Loadings cannot be found..")
}
loadings <- attr(tse, attr_name)
```

Reply:
I see but for example for RDA there is no attribute where it is stored, so you can't even retrieve the matrix that easily

Reply:
Sorry, last week I was busy with OMA, so I had not time to check this.

From RDA, you can get "constrained weighted orthonormal species scores" which are analogous to loadings of PCA. See for instance this https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/cca.object

When used getRDA, (getCCA) or add* functions, these are stored into "rotation" attribute of returned matrix.